lib/Frontend/OpenQASM3/OpenQASM3Frontend.cpp

```diff
@@ -202,7 +202,7 @@ llvm::Error qssc::frontend::openqasm3::parse(
         llvm::errs() << level << " while parsing OpenQASM 3 input\n"
                      << errMsg.str();
 
-        if (diagnosticCallback_) {
+        if (diagnosticCallback_ and diagLevel == qssc::Severity::Error) {
```

Comment:
Should we add other `diagLevel`s to any other kind of category

Reply:
Added ICE severity to diagnostics in https://github.com/openqasm/qe-compiler/pull/296/commits/9c6d6a5e2b17281426c27fdb4df63b791cfc727d